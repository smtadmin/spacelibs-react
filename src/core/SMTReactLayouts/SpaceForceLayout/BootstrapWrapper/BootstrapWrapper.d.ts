export default BootstrapWrapper;
declare class BootstrapWrapper {
    constructor(props: any);
    props: any;
    buildSections(): void;
    content: JSX.Element[];
    render(): JSX.Element[];
}
declare namespace BootstrapWrapper {
    namespace defaultProps {
        const bootstrap: string;
    }
    namespace propTypes {
        export const content: any;
        export const sectionName: any;
        const bootstrap_1: any;
        export { bootstrap_1 as bootstrap };
    }
}
