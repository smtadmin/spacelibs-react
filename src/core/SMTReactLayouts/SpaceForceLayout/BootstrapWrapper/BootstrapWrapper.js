/*
 * File: BootstrapWrapper
 * Version: 0.1.0
 * Project: react-test
 * Description: A React component that wraps whatever is passed to it in the specified bootstrap wrappers
 * File Created: Wednesday, 3rd February 2021 1:06 pm
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Wednesday, 10th February 2021 3:01 pm
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
import React from 'react';
import { Row } from 'react-bootstrap';
import PropTypes from 'prop-types';

/**
 * Class for wraping components in bootstrap wrappers
 */
class BootstrapWrapper extends React.Component {

    /**
     * Creates an instance of a bootstrap wrapper
     * @param {*} props The props recieved by the component
     */
    constructor(props) {
        super();
        this.props = props;
        this.buildSections();
    }

    /**
     * Checks wraps elements in divs and assigns each element a specific key and class so that they are easily manipulated using css
     * @param {*} props Object passed to the class
     */
    buildSections() {
        const newElements = [];
        let count = 0;
        if (Array.isArray(this.props.content)) {
            for (const element of this.props.content) {
                newElements.push(<div key={this.props.sectionName + "" + count} className={this.props.sectionName + "-header-element-" + count + " col"}>{element}</div>)
                count++;
            }
        } else {
            newElements.push(<div key={this.props.sectionName + "" + count} className={this.props.sectionName + "-header-element-" + count + " col"}>{this.props.content}</div>)
        }
        this.content = [<Row key={this.props.sectionName} className={this.props.sectionName + "-direction-wrapper " + this.props.bootstrap}>{newElements}</Row>];
    }

    /**
     * Renders the built content as a component and returns it
     */
    render() {
        return this.content;
    }

}


BootstrapWrapper.defaultProps = {
    bootstrap: ""
};
/** Prop Validators */
BootstrapWrapper.propTypes = {
    content: PropTypes.any,
    sectionName: PropTypes.string,
    bootstrap: PropTypes.string
};

export default BootstrapWrapper;