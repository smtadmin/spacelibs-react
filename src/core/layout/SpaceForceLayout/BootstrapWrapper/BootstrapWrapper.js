/*
 * File: BootstrapWrapper
 * Version: 0.1.0
 * Project: react-test
 * Description: A React component that wraps whatever is passed to it in the specified bootstrap wrappers
 * File Created: Wednesday, 3rd February 2021 1:06 pm
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Thursday, 18th March 2021 11:57 am
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
import React from 'react';
import Row from 'react-bootstrap/Row';
import PropTypes from 'prop-types';

/**
 * Class for wraping components in bootstrap wrappers
 */
class BootstrapWrapper extends React.Component {

    /**
     * Creates an instance of a bootstrap wrapper
     * @param {*} props The props recieved by the component
     */
    constructor(props) {
        super(props);
        this.buildSections();
    }

    /**
     * Checks wraps elements in divs and assigns each element a specific key and class so that they are easily manipulated using css
     */
    buildSections() {
        const newElements = [];
        let count = 0;
        if (Array.isArray(this.props.content)) {
            for (const element of this.props.content) {
                newElements.push(<div key={this.props.sectionName + "" + count} className={this.props.sectionName + "-header-element-" + count + " col"}>{element}</div>);
                count++;
            }
        } else {
            newElements.push(<div key={this.props.sectionName + "" + count} className={this.props.sectionName + "-header-element-" + count + " col"}>{this.props.content}</div>);
        }
        this.content = [<Row key={this.props.sectionName} className={this.props.sectionName + "-direction-wrapper " + this.props.bootstrap}>{newElements}</Row>];
    }

	/**
	 * Renders BootstrapWrapper
	 *
	 * @returns {*} React Components 
	 * @memberof BootstrapWrapper
	 */
	render() {
        return this.content;
    }

}

/** Sets Default Props */
BootstrapWrapper.defaultProps = {
    bootstrap: ""
};
/** Prop Validators */
BootstrapWrapper.propTypes = {
    content: PropTypes.any,
    sectionName: PropTypes.string,
    bootstrap: PropTypes.string
};

export default BootstrapWrapper;