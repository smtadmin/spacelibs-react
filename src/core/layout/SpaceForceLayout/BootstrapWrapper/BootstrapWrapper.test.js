/*
 * File: BootstrapWrapper.test
 * Version: 0.1.0
 * Project: react-test
 * Description: Testing Suite for the Bootstrap Wrapper Component
 * File Created: Saturday, 6th February 2021 10:17 am
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Wednesday, 10th February 2021 3:05 pm
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import BootstrapWrapper from './BootstrapWrapper';
import React from 'react';
import { render } from '@testing-library/react';

/**
 * Attempts to wrap a div without crashing
 */
it("Renders without crashing", () => {
    const element = render(<BootstrapWrapper sectionName="hello-world" content={<div>Hello World</div>} />)
    expect(element).toBeTruthy();
});


/**
 * Renders a div element and compares the html output to the stored snapshot
 */
it("Renders Correctly", () => {
    const element = render(<BootstrapWrapper sectionName="hello-world" content={<div>Hello World</div>} />)
    expect(element).toMatchSnapshot();
})

/**
 * Renders an array of divs and compares the html output to the stored snapshot
 */
it("Renders Multiple Elements Correctly", () => {
    const element = render(<BootstrapWrapper sectionName="hello-world" content={[<div>Element1</div>, <div>Element2</div>, <div>Element3</div>]} />)
    expect(element).toMatchSnapshot();
})
