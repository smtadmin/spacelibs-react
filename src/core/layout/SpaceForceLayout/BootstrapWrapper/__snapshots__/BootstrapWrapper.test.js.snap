// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hello-world-direction-wrapper  row"
      >
        <div
          class="hello-world-header-element-0 col"
        >
          <div>
            Hello World
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hello-world-direction-wrapper  row"
    >
      <div
        class="hello-world-header-element-0 col"
      >
        <div>
          Hello World
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders Multiple Elements Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hello-world-direction-wrapper  row"
      >
        <div
          class="hello-world-header-element-0 col"
        >
          <div>
            Element1
          </div>
        </div>
        <div
          class="hello-world-header-element-1 col"
        >
          <div>
            Element2
          </div>
        </div>
        <div
          class="hello-world-header-element-2 col"
        >
          <div>
            Element3
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hello-world-direction-wrapper  row"
    >
      <div
        class="hello-world-header-element-0 col"
      >
        <div>
          Element1
        </div>
      </div>
      <div
        class="hello-world-header-element-1 col"
      >
        <div>
          Element2
        </div>
      </div>
      <div
        class="hello-world-header-element-2 col"
      >
        <div>
          Element3
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
